priam/src/main/java/com/netflix/priam/backupv2/ForgottenFilesManager.java

```diff
@@ -125,22 +131,74 @@ public void findAndMoveForgottenFiles(Instant snapshotInstant, File snapshotDir)
+                        Paths.get(destDir.toFile().getAbsolutePath(), file.toFile().getName());
+                // Lets see if there is a symbolic link to this file already?
+                if (!Files.exists(symbolic_link)) {
+                    // If not, lets create one and work on next file.
+                    Files.createSymbolicLink(symbolic_link, file);
+                    continue;
+                } else if (Files.isSymbolicLink(symbolic_link)) {
+                    // Symbolic link exists, is it older than our timeframe?
+                    Instant last_modified_time =
+                            Files.getLastModifiedTime(symbolic_link, LinkOption.NOFOLLOW_LINKS)
+                                    .toInstant();
+                    if (DateUtil.getInstant()
+                            .isAfter(
+                                    last_modified_time.plus(
+                                            config.getForgottenFileGracePeriodDaysForRead(),
+                                            ChronoUnit.DAYS))) {
+                        // Eligible for move.
+                        logger.warn(
+                                "Forgotten file: {} found for CF: {}",
+                                file,
+                                columnfamilyDir.getName());
+                        backupMetrics.incrementForgottenFiles(1);
+                        if (config.isForgottenFileMoveEnabled()) {
+                            try {
+                                // Remove our symbolic link. Note that deletion of symbolic link
+                                // does not remove the original file.
+                                Files.delete(symbolic_link);
+                                FileUtils.moveFileToDirectory(
+                                        file.toFile(), destDir.toFile(), true);
+                            } catch (IOException e) {
+                                logger.error(
+                                        "Exception occurred while trying to move forgottenFile: {}. Ignoring the error and continuing with remaining backup/forgotten files.",
+                                        file);
+                                e.printStackTrace();
+                            }
+                        }
+                    }
                 }
+
+            } catch (IOException e) {
+                logger.error("Forgotten file: Error while trying to process the file: {}", file);
+            }
+        }
+
+        // Clean LOST_FOUND directory of any previous symbolic link files which are not considered
+        // lost any more.
+        for (File file : FileUtils.listFiles(destDir.toFile(), null, false)) {
+            Path filePath = Paths.get(file.getAbsolutePath());
+            if (Files.isSymbolicLink(filePath)) {
+                Path originalFile = Files.readSymbolicLink(filePath);
+                if (!columnfamilyPaths.contains(originalFile)) Files.delete(filePath);
```

Comment:
Might be a good idea to log a message here, this also indicates how many times we got into above mentioned C* edgecases.